freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,252 @@
+                    for col_name, values in compare_df.items():
+                        col_idx = compare_df.columns.get_loc(col_name)
+                        compare_df_row = compare_df.iloc[0]
+                        # compare_df now comprises tuples with [1] having either 'self' or 'other'
+                        if 'other' in col_name[1]:
+                            continue
+                        self_value = compare_df_row[col_idx]
+                        other_value = compare_df_row[col_idx + 1]
+
+                        # output differences
+                        if self_value != other_value:
+
+                            if not self.current_analysis.false_indicators.__contains__(col_name[0]):
+                                self.current_analysis.false_indicators.append(col_name[0])
+                                print(f"=> found look ahead bias in indicator {col_name[0]}. " +
+                                      f"{str(self_value)} != {str(other_value)}")
+
+    def prepare_data(self, varHolder, pairs_to_load):
+
+        # purge previous data
+        abs_folder_path = pathlib.Path("user_data/models/uniqe-id").resolve()
+        # remove folder and its contents
+        if pathlib.Path.exists(abs_folder_path):
+            shutil.rmtree(abs_folder_path)
+
+        prepare_data_config = copy.deepcopy(self.local_config)
+        prepare_data_config['timerange'] = (str(self.dt_to_timestamp(varHolder.from_dt)) + "-" +
+                                            str(self.dt_to_timestamp(varHolder.to_dt)))
+        prepare_data_config['exchange']['pair_whitelist'] = pairs_to_load
+
+        self.backtesting = Backtesting(prepare_data_config)
+        self.backtesting._set_strategy(self.backtesting.strategylist[0])
+        varHolder.data, varHolder.timerange = self.backtesting.load_bt_data()
+        self.backtesting.load_bt_data_detail()
+
+        varHolder.indicators = self.backtesting.strategy.advise_all_indicators(varHolder.data)
+        varHolder.result = self.get_result(self.backtesting, varHolder.indicators)
+
+    def start(self, config, strategy_obj: dict, args) -> None:
+
+        # deepcopy so we can change the pairs for the 2ndary runs
+        # and not worry about another strategy to check after.
+        self.local_config = deepcopy(config)
+        self.local_config['strategy_list'] = [strategy_obj['name']]
+        self.current_analysis = Analysis()
+        self.minimum_trade_amount = args['minimum_trade_amount']
+        self.targeted_trade_amount = args['targeted_trade_amount']
+        self.exportfilename = args['exportfilename']
+        self.strategy_obj = strategy_obj
+
+        # first make a single backtest
+        self.full_varHolder = VarHolder()
+
+        # define datetime in human-readable format
+        parsed_timerange = TimeRange.parse_timerange(config['timerange'])
+
+        if parsed_timerange.startdt is None:
+            self.full_varHolder.from_dt = datetime.utcfromtimestamp(0)
```

Comment:
please don't use this method, it's simply wrong.
it returns a 0 timestamp in utc - but automatically converts it to local time.
This can be no problem - but can also fail if you're in a negative offset timezone.

instead, `datetime.fromtimestamp(0, tz=timezone.utc)` should be used.

Reply:
sure